mev-boost/server/constraints.go

```diff
@@ -7,22 +7,25 @@ import (
 	lru "github.com/hashicorp/golang-lru/v2"
 )
 
-type BatchedSignedConstraints = []*SignedConstraints
+type (
+	BatchedSignedConstraints = []*SignedConstraints
+	HashToTransactionDecoded = map[gethCommon.Hash]*types.Transaction
+)
 
+// SignedConstraints represents the signed constraints.
+// Reference: https://docs.boltprotocol.xyz/api/builder
 type SignedConstraints struct {
 	Message   ConstraintsMessage  `json:"message"`
 	Signature phase0.BLSSignature `json:"signature"`
 }
 
+// ConstraintsMessage represents the constraints message.
+// Reference: https://docs.boltprotocol.xyz/api/builder
 type ConstraintsMessage struct {
-	ValidatorIndex uint64        `json:"validator_index"`
-	Slot           uint64        `json:"slot"`
-	Constraints    []*Constraint `json:"constraints"`
-}
-
-type Constraint struct {
-	Tx    Transaction `json:"tx"`
-	Index *uint64     `json:"index"`
+	Pubkey       phase0.BLSPubKey `json:"pubkey"`
+	Slot         uint64           `json:"slot"`
+	Top          bool             `json:"top"`
+	Transactions []Transaction    // Custom marshal and unmarshal implemented below
```

Comment:
```suggestion
	Transactions []*Transaction    `json:"transactions"`
```
You probably don't need to do additional encoding and decoding here so the concrete transaction type with custom marshal and unmarshal done here https://github.com/chainbound/bolt/blob/naman%2Ffeat%2Fmev-boost%2Fconstraints-api/builder/core/types/constraints.go#L178-L230 is probably not needed.
However you still have to specify the name of the field otherwise the error:
```text
time="2024-10-07T15:00:26.088Z" level=info msg="sent request for 1 constraint to relay. err = HTTP error response: 400 / Serde decode error: missing field `transactions` at line 1 column 378" method=submitConstraint ua=bolt-sidecar url="http://172.16.0.21:4040/constraints/v1/builder/constraints" version=
```

is thrown